internal/handler/wallet/sign.go

```diff
@@ -340,102 +395,76 @@ func (s *walletSign) prepareplainTextPromptRequest(
-// 				Code:    errorSignLoadCache,
-// 				Message: fmt.Sprintf("Error cannot unprotect from cache: %v", err.Error()),
-// 			})
-// 	}
-
-// 	return nil
-// }
-
-func CacheKey(correlationId models.CorrelationID) [32]byte {
-	return blake3.Sum256(correlationId)
+// privateKeyFromCache return the private key from the cache or an error.
+func (w *walletSign) privateKeyFromCache(
+	acc *account.Account,
+	correlationID models.CorrelationID,
+) (*memguard.LockedBuffer, error) {
+	key := CacheKey(correlationID)
+
+	value, err := w.gc.Get(key)
+	if err != nil {
+		if err.Error() == gcache.KeyNotFoundError.Error() {
+			return nil, fmt.Errorf("%w: %w", utils.ErrCorrelationIDNotFound, err)
+		}
+
+		return nil, fmt.Errorf("%w: %w", utils.ErrCache, err)
+	}
+
+	// convert interface{} into byte[]
+	buf := new(bytes.Buffer)
+
+	err = binary.Write(buf, binary.LittleEndian, value)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", utils.ErrCache, err)
+	}
+
+	cipheredPrivateKey := buf.Bytes()
+
+	privateKey, err := Xor(cipheredPrivateKey, correlationID)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", utils.ErrCache, err)
+	}
+
+	return memguard.NewBufferFromBytes(privateKey), nil
+}
+
+func Xor(a, b []byte) ([]byte, error) {
+	if len(a) != len(b) {
+		return nil, fmt.Errorf("length of two arrays must be same, %d and %d", len(a), len(b))
+	}
+	result := make([]byte, len(a))
+
+	for i := 0; i < len(a); i++ {
+		result[i] = a[i] ^ b[i]
+	}
+
+	return result, nil
 }
 
-// func handleBatch(
-// 	acc *account.Account,
-// 	params operations.SignParams,
-// 	s *walletSign,
-// 	gc gcache.Cache,
-// ) (models.CorrelationID, middleware.Responder) {
-// 	correlationId, err := generateCorrelationId()
-// 	if err != nil {
-// 		return nil, operations.NewSignInternalServerError().WithPayload(
-// 			&models.Error{
-// 				Code:    errorSignGenerateCorrelationId,
-// 				Message: fmt.Sprintf("Error cannot generate correlation id: %v", err.Error()),
-// 			})
-// 	}
-
-// 	key := CacheKey(correlationId)
-
-// 	cacheValue, err := wallet.Xor(acc.KeyPair.PrivateKey, correlationId)
-// 	if err != nil {
-// 		return nil, operations.NewSignInternalServerError().WithPayload(
-// 			&models.Error{
-// 				Code:    errorSignGenerateCorrelationId,
-// 				Message: fmt.Sprintf("Error cannot XOR correlation id: %v", err.Error()),
-// 			})
-// 	}
-
-// 	err = gc.SetWithExpire(key, cacheValue, passwordExpirationTime)
-// 	if err != nil {
-// 		return nil, operations.NewSignInternalServerError().WithPayload(
-// 			&models.Error{
-// 				Code:    errorSignGenerateCorrelationId,
-// 				Message: fmt.Sprintf("Error set correlation id in cache: %v", err.Error()),
-// 			})
-// 	}
-
-// 	return correlationId, nil
-// }
+func CacheKey(correlationID models.CorrelationID) [32]byte {
+	return blake3.Sum256(correlationID)
+}
+
+func expirationDuration() time.Duration {
+	fromEnv := os.Getenv("BATCH_EXPIRATION_TIME")
+
+	if fromEnv == "" {
+		return passwordExpirationTime
```

Comment:
Why is it named password ? You should also add a "default" somewhere in its name